File: client/src/models/LibraryWithBookViewModel.ts
import { z } from 'zod';
import AddressViewModel from './AddressViewModel';

const LibraryWithBookViewModel = z.object({
  id: z.string().uuid(),
  name: z.string(),
  latitude: z.number(),
  longitude: z.number(),
  address: AddressViewModel,
  isBookCurrentlyAvailable: z.boolean(),
});

export default LibraryWithBookViewModel;
export type LibraryWithBookViewModelType = z.infer<typeof LibraryWithBookViewModel>;

File: Server/src/Server.Application/ViewModels/LibraryWithBookViewModel.cs
namespace Server.Application.ViewModels;

public class LibraryWithBookViewModel
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public double Longitude { get; set; }
    public double Latitude { get; set; }
    public AddressViewModel Address { get; set; }
    public bool IsBookCurrentlyAvailable { get; set; }
}
